### PR TITLE
DEP: add deprecation warning for kwargs nyq / Hz in firwin

### LINKED ISSUE #16253: DEP: add deprecation warning for kwargs `nyq` / `Hz` in firwin*
Deprecated in documentation only since #7814.

### PR DESCRIPTION
**Pull request by j-bowhay**
#### Reference issue

closes gh-16253
#### What does this implement/fix?

Adds deprecation warning to kwargs `nyq` and `Hz` that were previously deprecated in docstring only

### CODE REVIEW

File: scipy/signal/_fir_filter_design.py [hunk 1]
```diff
@@ -880,10 +899,14 @@ def firls(numtaps, bands, desired, weight=None, nyq=None, fs=None):
...
-        Nyquist frequency. Each frequency in `bands` must be between 0
-        and `nyq` (inclusive). Default is 1.
+    nyq : float, optional, deprecated
+        This is the Nyquist frequency. Each frequency in `bands` must be
+        between 0 and `nyq` (inclusive). Default is 1.
+
+        .. Deprecated:: 1.0.0
+
+        `firls` keyword argument `nyq` is deprecated in favour of `fs` and
+        will be removed in SciPy 1.11.0.
```
Comment (tupui) [inline scipy/signal/_fir_filter_design.py:909]:
```suggestion
        .. Deprecated:: 1.0.0
           `firls` keyword argument `nyq` is deprecated in favour of `fs` and
           will be removed in SciPy 1.11.0.
```

**Review by tupui — CHANGES_REQUESTED** [review]
Thank @j-bowhay. A few suggestions

File: scipy/signal/tests/test_fir_filter_design.py [hunk 2]
```diff
@@ -247,8 +248,10 @@ def test_bad_cutoff(self):
         # 2D array not allowed.
         assert_raises(ValueError, firwin, 99, [[0.1, 0.2],[0.3, 0.4]])
         # cutoff values must be less than nyq.
-        assert_raises(ValueError, firwin, 99, 50.0, nyq=40)
-        assert_raises(ValueError, firwin, 99, [10, 20, 30], nyq=25)
+        with np.testing.suppress_warnings() as sup:
+            sup.filter(DeprecationWarning, "Keyword argument 'nyq'")
+            assert_raises(ValueError, firwin, 99, 50.0, nyq=40)
+            assert_raises(ValueError, firwin, 99, [10, 20, 30], nyq=25)
```
[the remark this answers is not in the document]
Comment (j-bowhay) [reply scipy/signal/tests/test_fir_filter_design.py:254]: Lines 255-256 perform the same test using fs so I think this is fine to have as well. Happy to change if you disagree

File: scipy/signal/_fir_filter_design.py [hunk 3]
```diff
@@ -310,10 +313,14 @@ def firwin(numtaps, cutoff, width=None, window='hamming', pass_zero=True,
...
-        Each frequency in `cutoff` must be between 0 and `nyq`. Default
-        is 1.
+    nyq : float, optional, deprecated
+        This is the Nyquist frequency. Each frequency in `cutoff` must be
+        between 0 and `nyq`. Default is 1.
+
+        .. Deprecated:: 1.0.0
+
+           `firwin` keyword argument `nyq` is deprecated in favour of `fs` and
+           will be removed in SciPy 1.11.0.
```
Comment (tupui) [inline scipy/signal/_fir_filter_design.py:323]:
```suggestion
        .. deprecated:: 1.0.0
           `firwin` keyword argument `nyq` is deprecated in favour of `fs` and
           will be removed in SciPy 1.11.0.
```

File: scipy/signal/_fir_filter_design.py [hunk 4]
```diff
@@ -519,9 +526,14 @@ def firwin2(numtaps, freq, gain, nfreqs=None, window='hamming', nyq=None,
...
-        *Deprecated. Use `fs` instead.* This is the Nyquist frequency.
-        Each frequency in `freq` must be between 0 and `nyq`.  Default is 1.
+    nyq : float, optional, deprecated
+        This is the Nyquist frequency. Each frequency in `freq` must be
+        between 0 and `nyq`. Default is 1.
+
+        .. Deprecated:: 1.0.0
+
+           `firwin2` keyword argument `nyq` is deprecated in favour of `fs` and
+           will be removed in SciPy 1.11.0.
```
Comment (tupui) [inline scipy/signal/_fir_filter_design.py:536]:
```suggestion
        .. deprecated:: 1.0.0
           `firwin2` keyword argument `nyq` is deprecated in favour of `fs` and
           will be removed in SciPy 1.11.0.
```

File: scipy/signal/_fir_filter_design.py [hunk 5]
```diff
@@ -707,9 +719,13 @@ def remez(numtaps, bands, desired, weight=None, Hz=None, type='bandpass',
...
         `weight` has to be half the length of `bands`.
-    Hz : scalar, optional
-        *Deprecated.  Use `fs` instead.*
+    Hz : scalar, optional, deprecated
         The sampling frequency in Hz. Default is 1.
+
+        .. Deprecated:: 1.0.0
+
+           `remez` keyword argument `Hz` is deprecated in favour of `fs` and
+           will be removed in SciPy 1.11.0.
```
Comment (tupui) [inline scipy/signal/_fir_filter_design.py:728]:
```suggestion
        .. deprecated:: 1.0.0
           `remez` keyword argument `Hz` is deprecated in favour of `fs` and
           will be removed in SciPy 1.11.0.
```

File: scipy/signal/_fir_filter_design.py [hunk 6]
```diff
@@ -880,10 +899,14 @@ def firls(numtaps, bands, desired, weight=None, nyq=None, fs=None):
...
-        Nyquist frequency. Each frequency in `bands` must be between 0
-        and `nyq` (inclusive). Default is 1.
+    nyq : float, optional, deprecated
+        This is the Nyquist frequency. Each frequency in `bands` must be
+        between 0 and `nyq` (inclusive). Default is 1.
+
+        .. Deprecated:: 1.0.0
+
+           `firls` keyword argument `nyq` is deprecated in favour of `fs` and
+           will be removed in SciPy 1.11.0.
```
Comment (tupui) [inline scipy/signal/_fir_filter_design.py:909]:
```suggestion
        .. deprecated:: 1.0.0
           `firls` keyword argument `nyq` is deprecated in favour of `fs` and
           will be removed in SciPy 1.11.0.
```

**Review by tupui — COMMENTED** [review]
This should fix the CI

**Comment by j-bowhay** [comment]
@tupui thanks for that, I keep messing that formatting up! I realise now that the formatting in #16265 and #16258 is incorrect should I make a flow up pr?

**Comment by tupui** [comment]
No worries 😃 It's not a priority but feel free. If you do, you could address all occurrences in the doc (not that many).

**Comment by larsoner** [comment]
@j-bowhay can you merge with `upstream/main` to see if that fixes CIs? It would be nice to be able to verify things are correct in the doc build

File: scipy/signal/_fir_filter_design.py [hunk 7]
```diff
@@ -310,10 +313,13 @@ def firwin(numtaps, cutoff, width=None, window='hamming', pass_zero=True,
...
 
-    nyq : float, optional
-        *Deprecated. Use `fs` instead.* This is the Nyquist frequency.
-        Each frequency in `cutoff` must be between 0 and `nyq`. Default
-        is 1.
+    nyq : float, optional, deprecated
+        This is the Nyquist frequency. Each frequency in `cutoff` must be
+        between 0 and `nyq`. Default is 1.
+
+        .. Deprecated:: 1.0.0
```
Comment (larsoner) [inline scipy/signal/_fir_filter_design.py:320]: Looks like these still need to be lower-cased

**Comment by larsoner** [comment]
According to CircleCI some examples need to be updated
```
/home/circleci/repo/build-install/lib/python3.8/site-packages/scipy/signal/_fir_filter_design.py:docstring of scipy.signal._fir_filter_design.kaiserord:67: WARNING: Exception occurred in plotting scipy-signal-kaiserord-1
 from /home/circleci/repo/doc/source/reference/generated/scipy.signal.kaiserord.rst:
Traceback (most recent call last):
  File "/home/circleci/.pyenv/versions/3.8.13/lib/python3.8/site-packages/matplotlib/sphinxext/plot_directive.py", line 517, in _run_code
    exec(code, ns)
  File "<string>", line 36, in <module>
  File "/home/circleci/repo/build-install/lib/python3.8/site-packages/scipy/signal/_fir_filter_design.py", line 393, in firwin
    nyq = 0.5 * _get_fs(fs, nyq)
  File "/home/circleci/repo/build-install/lib/python3.8/site-packages/scipy/signal/_fir_filter_design.py", line 31, in _get_fs
    warnings.warn(msg, DeprecationWarning, stacklevel=3)
DeprecationWarning: Keyword argument 'nyq' is deprecated in favour of 'fs' and will be removed in SciPy 1.12.0.
/home/circleci/repo/build-install/lib/python3.8/site-packages/scipy/signal/_fir_filter_design.py:docstring of scipy.signal._fir_filter_design.minimum_phase:102: WARNING: Exception occurred in plotting scipy-signal-minimum_phase-1
 from /home/circleci/repo/doc/source/reference/generated/scipy.signal.minimum_phase.rst:
Traceback (most recent call last):
  File "/home/circleci/.pyenv/versions/3.8.13/lib/python3.8/site-packages/matplotlib/sphinxext/plot_directive.py", line 517, in _run_code
    exec(code, ns)
  File "<string>", line 7, in <module>
  File "/home/circleci/repo/build-install/lib/python3.8/site-packages/scipy/signal/_fir_filter_design.py", line 854, in remez
    warnings.warn(msg, DeprecationWarning, stacklevel=2)
DeprecationWarning: 'remez' keyword argument 'Hz' is deprecated in favour of 'fs' and will be removed in SciPy 1.12.0.
/home/circleci/repo/build-install/lib/python3.8/site-packages/scipy/signal/_fir_filter_design.py:docstring of scipy.signal._fir_filter_design.remez:97: WARNING: Exception occurred in plotting scipy-signal-remez-1
 from /home/circleci/repo/doc/source/reference/generated/scipy.signal.remez.rst:
Traceback (most recent call last):
  File "/home/circleci/.pyenv/versions/3.8.13/lib/python3.8/site-packages/matplotlib/sphinxext/plot_directive.py", line 517, in _run_code
    exec(code, ns)
  File "<string>", line 29, in <module>
  File "/home/circleci/repo/build-install/lib/python3.8/site-packages/scipy/signal/_fir_filter_design.py", line 854, in remez
    warnings.warn(msg, DeprecationWarning, stacklevel=2)
DeprecationWarning: 'remez' keyword argument 'Hz' is deprecated in favour of 'fs' and will be removed in SciPy 1.12.0.
```

**Comment by larsoner** [comment]
(or the deprecation warning is not being emitted correctly / under the correct conditions -- not sure which is the case)

**Comment by j-bowhay** [comment]
@larsoner docs now rendering correctly, CI failings seem unrelated?

**Review by larsoner — APPROVED** [review]
@tupui feel free to merge if you're happy

**Review by tupui — APPROVED** [review]
LGTM, thanks everyone 😃

**Pull request merged**